tests/src/test_environnement.cpp

```diff
@@ -8,39 +9,43 @@ TEST(Environnment, test_constructor) {
 
     EXPECT_NEAR(e.left(), -100, 0.00001);
     EXPECT_NEAR(e.right(), 100, 0.00001);
-    EXPECT_NEAR(e.bottom(), -100, 0.00001);
-    EXPECT_NEAR(e.top(), 100, 0.00001);
+    EXPECT_NEAR(e.top(), -100, 0.00001);
+    EXPECT_NEAR(e.bottom(), 100, 0.00001);
 }
 
 TEST(Environnment, test_constructor_invalid_behavior) {
     // Creates an environment of size 200m x 200m but with wrong
-    // values.
-    Environment e{100,-100,-100,100};
-
-    EXPECT_GT(e.left(), e.right());
-    EXPECT_GT(e.top(), e.bottom());
+    // catch the exception
+    try {
+        Environment e{100, -100, -100, 100};
+        FAIL() << "Expected InvalidBoundariesEnvironmentException";
+    } catch (const InvalidBoundariesEnvironmentException &e) {
+        std::cout << e.what() << std::endl;  // Test passed, we caught the exception
+    } catch (...) {
+        FAIL() << "Expected InvalidBoundariesEnvironmentException";
+    }
+    
 }
 
 TEST(Environnment, test_add_element) {
-    // Creates an environment of size 200m x 200m but with wrong values.
-    Environment e{100,-100,-100,100};
-
-    //EXPECT_EQ(e.get_elements(), 0);
-    //e.add_element(new Element());
-    //e.add_element(new Element());
-    //EXPECT_EQ(e.get_elements(), 2);
+    // Creates an environment of size 200m x 200m with right values.
+    Environment e{-100,100,-100,100};
 
-    EXPECT_TRUE(false) << "Uncomment the line before";
+    EXPECT_EQ(e.get_elements(), 0);
+    e.add_element(new Element());
+    e.add_element(new Element());
+    EXPECT_EQ(e.get_elements(), 2);
 }
 
 TEST(Environnment, test_add_robot) {
-    // Creates an environment of size 200m x 200m but with wrong values.
-    Environment e{100,-100,-100,100};
+    // Creates an environment of size 200m x 200m with right values.
+    Environment e{-100,100,-100,100};
 
-    //EXPECT_EQ(e.get_objects(), 0);
-    //e.add_element(new Robot());
-    //e.add_element(new Robot());
-    //EXPECT_EQ(e.get_elements(), 2);
+    EXPECT_EQ(e.get_elements(), 0);
+    e.add_element(new Robot());
+    Robot* r = new Robot();
+    e.add_element(r);
+    EXPECT_EQ(e.get_elements(), 2);
+    EXPECT_EQ(r->get_environment(), &e);
```

Comment:
There is several memory leak in the code because of the memory allocated by the "new Robots()" which are not deleted. 

You can either:
- have a delete associated with every new.
- use an appropriate smart pointer that will handle the new/delete. 
- do not use memory allocated on the heap and use the stack instead as in:
```cpp
void aFunction(){
      Robot a{};       /// the memory holding the robot is in the stack 
                               /// the memory is released when the "a" instance is destroyed, this means at the end of the current scope.
                               /// between the a, and the '}' the object is valid
        
}  // << this is where the a instance is destroyed


```